### PR TITLE
Updated the code to use hc facilityId when applying facility on PO details page(#2admaej)

### PR DESCRIPTION
**Pull request by disha1202**
Description:
Updated the code to use hc facilityId when applying facility on PO details page
Additional change:
Handled the case when order state is empty on orderDetails page.

### CODE REVIEW

File: src/views/PurchaseOrder.vue [hunk 1]
```diff
@@ -126,14 +126,17 @@ export default defineComponent({
           shopifyProductUPC: [],
           arrivalDate: [],
           quantityOrdered: [],
-          facilityId: []
+          facilityId: [],
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:129]: I think we should set the `facilityId` and `externalFacilityId` as is in save method and rather manipulate in apply method only

File: src/views/OrderDetail.vue [hunk 2]
```diff
@@ -354,7 +360,9 @@ export default defineComponent({
       this.store.dispatch('order/updatedOrderListItems', this.ordersList.items);
     },
     getGroupList (items: any) {
-      return Array.from(new Set(items.map((ele: any) => ele.parentProductId)))
+      if (items.length) {
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:363]: Why this is needed?

File: src/views/PurchaseOrder.vue (hunk 1 above)
Comment (disha1202) [reply src/views/PurchaseOrder.vue:129]: Updated the code.

File: src/views/OrderDetail.vue (hunk 2 above)
Comment (disha1202) [reply src/views/OrderDetail.vue:360]: In save method, after uploading the data we clear the state and navigate to purchase order page. The order detail component is not unmounted and as the state is empty it results in an error.

File: src/views/OrderDetail.vue [hunk 3]
```diff
@@ -246,17 +247,18 @@ export default defineComponent({
       const uploadData = this.ordersList.items.filter((item: any) => {
         return item.isSelected;
       }).map((item: any) => {
-        return {
+        const orderItem = {
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:250]: We could directly return. WDYT?

File: src/views/OrderDetail.vue (hunk 2 above)
Comment (adityasharma7) [reply src/views/OrderDetail.vue:360]: I think if we are resetting items to [], we should not have error in any case and this handling is not required
Also, why not?
```
return [ ...new Set(items.map((ele: any) => ele.parentProductId))]
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/OrderDetail.vue (hunk 3 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Yes sir, updated the code to return order item object directly.

File: src/views/PurchaseOrder.vue [hunk 4]
```diff
@@ -126,14 +126,17 @@ export default defineComponent({
           shopifyProductUPC: [],
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:126]: I think above code could be refactored like this
```
this.orderItemsList = this.content.map(item => {
          return {
            orderId: item[this.orderIdField],
            shopifyProductSKU: item[this.productSkuField],
            shopifyProductUPC: item[this.productUpcField],
            arrivalDate: item[this.dateField],
            quantityOrdered: item[this.quantityField],
            facilityId: '',
            externalFacilityId: item[this.facilityField]
          }
        })
        ```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**